src/ledger/SyncingLedgerChain.cpp

```diff
@@ -33,21 +33,28 @@ SyncingLedgerChain::pop()
 SyncingLedgerChainAddResult
 SyncingLedgerChain::push(LedgerCloseData lcd)
 {
-    if (mChain.empty() ||
-        mChain.back().getLedgerSeq() + 1 == lcd.getLedgerSeq())
+    if (mLastLedgerSeq == 0 || mLastLedgerSeq + 1 == lcd.getLedgerSeq())
     {
         mChain.emplace(lcd);
+        mLastLedgerSeq = lcd.getLedgerSeq();
         return SyncingLedgerChainAddResult::CONTIGUOUS;
     }
 
-    if (lcd.getLedgerSeq() <= mChain.back().getLedgerSeq())
+    if (lcd.getLedgerSeq() <= mLastLedgerSeq)
     {
         return SyncingLedgerChainAddResult::TOO_OLD;
     }
 
     return SyncingLedgerChainAddResult::TOO_NEW;
 }
 
+void
+SyncingLedgerChain::reset()
+{
+    assert(empty());
```

Comment:
I don't understand the contract: why do we require `empty()`?
Also, this is probably more of a `runtime_error` type of situation as it's not a bug in the `SyncingLedgerChain` implementation, but a bug in what callers are doing.

Reply:
If this is not empty, it mean that we have not used all of the buffered ledgers, which mean that the catchup has not finished.